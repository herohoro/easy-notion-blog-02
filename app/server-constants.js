/* eslint @typescript-eslint/no-var-requires: 0 */
// use commonjs so it can be required without transpiling

const NOTION_API_SECRET = process.env.NOTION_API_SECRET
const DATABASE_ID = process.env.DATABASE_ID
const INDEX_PAGE_ID = process.env.INDEX_PAGE_ID
const MAP_PAGE_ID = process.env.MAP_PAGE_ID
const NEXT_PUBLIC_URL = process.env.NEXT_PUBLIC_URL
const NEXT_PUBLIC_GA_TRACKING_ID = process.env.NEXT_PUBLIC_GA_TRACKING_ID
const NEXT_PUBLIC_SITE_TITLE = 'herohoroブログ'
const NEXT_PUBLIC_SITE_DESCRIPTION =
  '非エンジニアがeasy-notion-blogを通して勉強しながらスキルアップをしていくブログ'
const NUMBER_OF_POSTS_PER_PAGE = 10
const SEC_NOTION_API_SECRET = process.env.SEC_NOTION_API_SECRET
const SEC_DATABASE_ID = process.env.SEC_DATABASE_ID

module.exports = {
  NOTION_API_SECRET,
  DATABASE_ID,
  INDEX_PAGE_ID,
  MAP_PAGE_ID,
  NEXT_PUBLIC_URL,
  NEXT_PUBLIC_GA_TRACKING_ID,
  NEXT_PUBLIC_SITE_TITLE,
  NEXT_PUBLIC_SITE_DESCRIPTION,
  NUMBER_OF_POSTS_PER_PAGE,
  SEC_NOTION_API_SECRET,
  SEC_DATABASE_ID,
}
