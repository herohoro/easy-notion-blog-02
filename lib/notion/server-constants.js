/* eslint @typescript-eslint/no-var-requires: 0 */
// use commonjs so it can be required without transpiling

const path = require('path')

const NOTION_API_SECRET = process.env.NOTION_API_SECRET
const DATABASE_ID = process.env.DATABASE_ID
const INDEX_PAGE_ID = process.env.INDEX_PAGE_ID
const MAP_PAGE_ID = process.env.MAP_PAGE_ID
const NEXT_PUBLIC_URL = process.env.NEXT_PUBLIC_URL
const NEXT_PUBLIC_GA_TRACKING_ID = process.env.NEXT_PUBLIC_GA_TRACKING_ID
const NEXT_PUBLIC_SITE_TITLE = 'herohoroブログ'
const NEXT_PUBLIC_SITE_DESCRIPTION =
  '非エンジニアがeasy-notion-blogを通して勉強しながらスキルアップをしていくブログ'
const BLOG_INDEX_CACHE = path.resolve('.blog_index_data')
const NUMBER_OF_POSTS_PER_PAGE = 10
const SEC_NOTION_API_SECRET = process.env.SEC_NOTION_API_SECRET
const SEC_DATABASE_ID = process.env.SEC_DATABASE_ID

module.exports = {
  NOTION_API_SECRET,
  DATABASE_ID,
  INDEX_PAGE_ID,
  MAP_PAGE_ID,
  NEXT_PUBLIC_URL,
  NEXT_PUBLIC_GA_TRACKING_ID,
  NEXT_PUBLIC_SITE_TITLE,
  NEXT_PUBLIC_SITE_DESCRIPTION,
  BLOG_INDEX_CACHE,
  NUMBER_OF_POSTS_PER_PAGE,
  SEC_NOTION_API_SECRET,
  SEC_DATABASE_ID,
}
