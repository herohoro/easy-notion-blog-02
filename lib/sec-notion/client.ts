import {
  SEC_NOTION_API_SECRET,
  SEC_DATABASE_ID,
} from '../../app/server-constants'
import { Post } from './interfaces'

const { Client } = require('@notionhq/client')
const client = new Client({
  auth: SEC_NOTION_API_SECRET,
})

export async function getSecShinyaMessage() {
  let results = []

  const params = {
    database_id: SEC_DATABASE_ID,
    filter: {
      property: 'tags',
      select: {
        equals: '深夜学習_つぶやき',
      },
    },
    sorts: [
      {
        property: 'last_edit',
        direction: 'descending',
      },
    ],
  }
  const data = await client.databases.query(params)

  results = results.concat(data.results)

  params['start_cursor'] = data.next_cursor

  return results.map((item) => _buildPost(item))
}

export async function getAllSecShinyaPosts() {
  let results = []

  const params = {
    database_id: SEC_DATABASE_ID,
    filter: {
      property: 'tags',
      select: {
        equals: '深夜学習',
      },
    },
    sorts: [
      {
        property: 'date',
        direction: 'descending',
      },
    ],
  }
  const data = await client.databases.query(params)

  results = results.concat(data.results)

  params['start_cursor'] = data.next_cursor

  return results.map((item) => _buildPost(item))
}

export async function getAllSecPosts() {
  let results = []

  const params = {
    database_id: SEC_DATABASE_ID,
    filter: {
      property: 'tags',
      select: {
        equals: 'posting',
      },
    },
    sorts: [
      {
        property: 'date',
        direction: 'descending',
      },
    ],
  }
  const data = await client.databases.query(params)

  results = results.concat(data.results)

  params['start_cursor'] = data.next_cursor

  return results.map((item) => _buildPost(item))
}

function _buildPost(data) {
  const prop = data.properties

  const post: Post = {
    title: prop.title.title[0].plain_text,
    date: prop.date.date.start,
    tags: prop.tags.select.name,
    description:
      prop.description.rich_text.length > 0
        ? prop.description.rich_text[0].plain_text
        : '',
    last_edit: prop.last_edit.last_edited_time,
    URL: prop.URL.url,
    site: prop.site.select.name,
    siteCollor: prop.site.select.color,
  }

  return post
}
