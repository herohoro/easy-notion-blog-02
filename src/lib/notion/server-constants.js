/* eslint @typescript-eslint/no-var-requires: 0 */
// use commonjs so it can be required without transpiling

const path = require('path')

const NOTION_API_SECRET = process.env.NOTION_API_SECRET
const DATABASE_ID = process.env.DATABASE_ID
const NEXT_PUBLIC_URL = process.env.NEXT_PUBLIC_URL
const NEXT_PUBLIC_GA_TRACKING_ID = process.env.NEXT_PUBLIC_GA_TRACKING_ID
const BLOG_INDEX_CACHE = path.resolve('.blog_index_data')
const NUMBER_OF_POSTS_PER_PAGE = 10
const SEC_NOTION_API_SECRET = process.env.SEC_NOTION_API_SECRET
const SEC_DATABASE_ID = process.env.SEC_DATABASE_ID

module.exports = {
  NOTION_API_SECRET,
  DATABASE_ID,
  NEXT_PUBLIC_URL,
  NEXT_PUBLIC_GA_TRACKING_ID,
  BLOG_INDEX_CACHE,
  NUMBER_OF_POSTS_PER_PAGE,
  SEC_NOTION_API_SECRET,
  SEC_DATABASE_ID
}
