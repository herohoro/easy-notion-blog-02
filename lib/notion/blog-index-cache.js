/* eslint @typescript-eslint/no-var-requires: 0 */
// use commonjs so it can be required without transpiling
import path from "node:path"
import fs from 'fs'
import { Client } from '@notionhq/client'
import { NOTION_API_SECRET, DATABASE_ID } from '../../app/server-constants.js'

const BLOG_INDEX_CACHE = path.resolve('.blog_index_data')

const notionClient = new Client({
  auth: NOTION_API_SECRET,
})

export function exists() {
  return fs.existsSync(BLOG_INDEX_CACHE)
}

export function get() {
  return JSON.parse(fs.readFileSync(BLOG_INDEX_CACHE))
}

export async function set() {
  let params = {
    database_id: DATABASE_ID,
    filter: {
      and: [
        {
          property: 'Published',
          checkbox: {
            equals: true,
          },
        },
        {
          property: 'Date',
          date: {
            on_or_before: new Date().toISOString(),
          },
        },
      ],
    },
    sorts: [
      {
        property: 'Date',
        timestamp: 'created_time',
        direction: 'descending',
      },
    ],
    page_size: 100,
  }

  let results = []

  while (true) {
    const data = await notionClient.databases.query(params)

    results = results.concat(data.results)

    if (!data.has_more) {
      break
    }

    params['start_cursor'] = data.next_cursor
  }

  fs.writeFileSync(BLOG_INDEX_CACHE, JSON.stringify(results))
  console.log(`Cached ${results.length} posts into ${BLOG_INDEX_CACHE}`)

  return
}

export function expire() {
  return fs.unlinkSync(BLOG_INDEX_CACHE)
}
