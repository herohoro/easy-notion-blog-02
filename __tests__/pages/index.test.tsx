import { render } from '@testing-library/react'
import RenderPage from '../../src/pages/index'
import { getAllBlocksByBlockId } from '../../src/lib/notion/client'
import { INDEX_PAGE_ID } from '../../src/lib/notion/server-constants'

jest.mock('next/router', () => ({
  useRouter() {
    return {
      asPath: '/',
      pathname: '/',
    }
  },
}))

describe('RenderPage', () => {
  it('renders the page unchanged', async () => {
    const blocks = await getAllBlocksByBlockId(INDEX_PAGE_ID)
    const { container } = render(<RenderPage blocks={blocks} />)
    expect(container).toMatchSnapshot()
  })
})
