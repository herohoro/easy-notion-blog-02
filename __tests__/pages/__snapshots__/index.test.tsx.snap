// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RenderPage renders the page unchanged 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="mainContent"
    >
      <div
        class="flexTagsMain"
      />
      <div
        style="display: flex; justify-content: center;"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27300%27%20height=%27300%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt=""
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain;"
          />
          <noscript />
        </span>
      </div>
      <a
        href="#Heading 1"
        id="Heading 1"
      >
        <h4>
          Heading 1
        </h4>
      </a>
      <a
        href="#Heading 2"
        id="Heading 2"
      >
        <h5>
          Heading 2
        </h5>
      </a>
      <a
        href="#Heading 3"
        id="Heading 3"
      >
        <h6>
          Heading 3
        </h6>
      </a>
      <p>
        Paragraph
      </p>
      <p>
        <b>
          bold
        </b>
      </p>
      <p>
        <i>
          italic
        </i>
      </p>
      <p>
        <u>
          Underline
        </u>
      </p>
      <p>
        <s>
          Strike
        </s>
      </p>
      <p>
        <a
          href="https://github.com/otoyo/easy-notion-blog"
        >
          link
        </a>
      </p>
      <p>
        <code>
          code
        </code>
      </p>
      <p />
      <ul>
        <li>
          Bulleted List
        </li>
        <li>
          Uni
        </li>
        <li>
          Tako
          <ul>
            <li>
              aaa
            </li>
            <li>
              bbb
            </li>
          </ul>
        </li>
      </ul>
      <p />
      <ol>
        <li>
          Numbered List
        </li>
        <li>
          Kani
        </li>
        <li>
          Ika
          <ol
            type="a"
          >
            <li>
              111
            </li>
            <li>
              222
            </li>
          </ol>
        </li>
      </ol>
      <p />
      <p>
        Divider
      </p>
      <hr
        class="divider"
      />
      <p />
      <div
        class="callout"
      >
        <div>
          ℹ️
        </div>
        <div>
          Callout
        </div>
      </div>
      <p />
      <p />
      <blockquote>
        Quote Block
      </blockquote>
      <p />
      <div
        class="table"
      >
        <table>
          <tbody>
            <tr>
              <th>
                Simple
              </th>
              <th>
                Table
              </th>
            </tr>
            <tr>
              <td>
                1
              </td>
              <td>
                Tako
              </td>
            </tr>
            <tr>
              <td>
                2
              </td>
              <td>
                Ika
              </td>
            </tr>
            <tr>
              <td>
                3
              </td>
              <td>
                Uni
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <p />
      <figure
        class="image"
      >
        <div>
          <img
            alt="画像が読み込まれない場合はページを更新してみてください。"
            src="https://s3.us-west-2.amazonaws.com/secure.notion-static.com/5d04be89-54da-482a-9391-efd465f74082/profile.png"
          />
        </div>
        <figcaption
          class="caption"
        >
          This is image caption.
        </figcaption>
      </figure>
      <div
        class="pcode"
      >
        <iframe
          height="100%"
          scrolling="no"
          src="https://p5-blog.vercel.app/%E3%83%89%E3%83%83%E3%83%88/index.html"
          width="100%"
        />
      </div>
    </div>
    <div
      class="subContent"
    >
      <div
        class="blogTagLink"
      >
        <h3>
          Category List
        </h3>
        <hr />
        <div
          class="noContents"
        >
          There are no contents yet
        </div>
      </div>
      <div
        class="blogTagLink"
      >
        <h3>
          Tag List
        </h3>
        <hr />
        <div
          class="noContents"
        >
          There are no contents yet
        </div>
      </div>
      <h3>
        Prolile
      </h3>
      <hr />
      <div
        class="flexWraper"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27200%27%20height=%27200%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt=""
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain;"
          />
          <noscript />
        </span>
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2780%27%20height=%2780%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt=""
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain;"
          />
          <noscript />
        </span>
        <ul>
          <li>
            勉強が趣味
          </li>
          <li>
            教えるの好き
          </li>
          <li>
            オンライン学習塾で５画面を操り指導(自称：職人)
          </li>
          <li>
            元教員・介護士
          </li>
          <li>
            家の中では無限大
          </li>
        </ul>
      </div>
      <h3>
        Study TimeLine
      </h3>
      <hr />
      <iframe
        height="400"
        src="https://notion2charts.com/embed/bad01964-6bce-4f62-bc9d-1d2899652ed6"
        width="100%"
      />
      <div
        class="blogPostLink"
      >
        <h3>
          Recommended
        </h3>
        <hr />
        <div
          class="noContents"
        >
          There are no contents yet
        </div>
      </div>
      <div
        class="twitterTimeline"
      >
        <h3>
          Twitter Timeline
        </h3>
        <hr />
        <p>
          フォロー大歓迎＼(^o^)／
        </p>
        <a
          class="twitter-timeline"
          data-chrome="nofooter,transparent,noheader"
          data-height="500"
          data-lang="en"
          data-theme="light"
          data-width="500"
          href="https://twitter.com/mineral_30?ref_src=twsrc%5Etfw"
        >
          Tweets by mineral_30
        </a>
         
        <script
          async=""
          src="https://platform.twitter.com/widgets.js"
        />
      </div>
    </div>
  </div>
</div>
`;
